contract/bridge/contracts/VaultRegistry.sol

```diff
@@ -393,5 +379,32 @@ abstract contract VaultRegistry is Initializable, ICollateral {
         // vault.toBeRedeemed = 0;
     }
 
-    uint256[45] private __gap;
+    function requireVaultExistence(uint256 _vaultBtcPublicKeyX) private {
+        require(_vaultBtcPublicKeyX != 0, "Vault does not exist");
+    }
+
+    function getVault(address _vaultId) external view override returns (uint256, uint256, uint256, uint256, uint256, uint256, uint256, uint256, uint256) {
+        Vault memory vault = vaults[_vaultId];
+
+        return (
+            vault.btcPublicKeyX,
+            vault.btcPublicKeyY,
+            vault.collateral,
+            vault.issued,
+            vault.toBeIssued,
+            vault.toBeRedeemed,
+            vault.replaceCollateral,
+            vault.toBeReplaced,
+            vault.liquidatedCollateral
+        );
+    }
+
+    function setVaultRewardAddress(address _vaultReward) external {
```

Comment:
On the note or architecture,  because we use inheritance, and expose various inherited classes as external, it be quite difficult to which is external layer, and internal layer.   I think with code size refactor later as we move away from inheritance, we can make this clearer.